### ISSUE TITLE
Use Tesla as wrapper to supports multiple http client

### ISSUE THREAD

**Issue by Matsa59** [codedge-llc/pigeon#233]
**Is your feature request related to a problem? Please describe.**

We moved from HTTPoison to hackney using Tesla but pigeon use directly HTTPoison create a dep that we could easily avoid.

**Describe the solution you'd like**

Replace HTTPoison by Tesla to support more http client.

After reading the code base it sounds really easy to do. I can do it but I prefer to ask if this direction is ok for the community.
At the end, people could continue to use HTTPoison if they prefer to

**Comment by hpopp** [comment]
I agree that migrating to Tesla would be really nice, but there's a lot more to consider than the current HTTPoison usage in Pigeon. If we were to make such a change, I would want to deprecate Kadabra and update FCM and APNS to use Tesla as well. This introduces a lot of other considerations (eg, Tesla would _have_ to use an HTTP/2-compliant adapter).

Let me think on it for a bit.

**Comment by hpopp** [comment]
v2.1.0 is going to ship with Mint managing everything, and the http adapter going away. Marking this closed.
